tensorboard/webapp/widgets/custom_modal/custom_modal.ts

```diff
@@ -109,23 +110,23 @@ export class CustomModal {
     const customModalRef = new CustomModalRef(overlayRef);
     this.customModalRefs.push(customModalRef);
 
-    // setTimeout to prevent closing immediately after modal open.
-    setTimeout(() => {
-      const outsidePointerEventsSubscription = overlayRef
-        .outsidePointerEvents()
-        .subscribe((event) => {
-          // Only close when click is outside of every modal
-          if (
-            this.customModalRefs.every(
-              (ref) =>
-                !isMouseEventInElement(event, ref.overlayRef.overlayElement)
-            )
-          ) {
-            this.closeAll();
-          }
-        });
-      customModalRef.subscriptions.push(outsidePointerEventsSubscription);
-    });
+    const outsidePointerEventsSubscription = overlayRef
+      .outsidePointerEvents()
+      .pipe(
+        skipUntil(timer(250)) // prevent closing immediately after modal open.
+      )
+      .subscribe((event) => {
```

Comment:
As I understand, the previous version of the code was executing immediately in some browsers (but apparently not in all of them), causing the event handler to be attached right before the "mouse up" event, or something like this, causing it to close it.

Can we instead check the type of event? Or something like this, to only match clicks? If an alternative solution is more complex, or difficult to validate, or if what I'm saying is not accurate or something, I'm ok with this solution. Just thought I'd ask.

Reply:
Oh this is better! I hadn't realized that there are two separate events for "right clicking": [contextmenu](https://developer.mozilla.org/en-US/docs/Web/API/Element/contextmenu_event) (immediately triggers the system context menu) and [auxclick](https://developer.mozilla.org/en-US/docs/Web/API/Element/auxclick_event) (triggers when mousedown and mouseup occurs in the same element)

Since [we're using contextmenu](https://github.com/tensorflow/tensorboard/blob/0627ad5480b309a5f3f69f39e52670496c2e5863/tensorboard/webapp/widgets/data_table/header_cell_component.ts#L48) to open the modal, I updated the code to simply ignore the auxclick events, which should have the effect of ignoring the right click mouseup. To close the modal, left clicks and contextmenu events can still be used.